invenio_app_ils/documents/indexer.py

```diff
@@ -63,9 +64,13 @@ def get_related_records(document_pid):
     """Get referenced records via relations."""
     referenced = []
     doc_record_cls = current_app_ils.document_record_cls
-    record = doc_record_cls.get_record_by_pid(document_pid)
+    try:
+        record = doc_record_cls.get_record_by_pid(document_pid)
+    except PIDDeletedError:
+        # Document might have been deleted while reindexing asyncronously.
+        return []
```

Comment:
nit: maybe it would be better to instead of allocating memory for a new list
```suggestion
        return referenced
```